### DISCUSSION TITLE
Brainstorming ideas

### DISCUSSION THREAD

**Discussion by cioraneanu** [cioraneanu/firefly-pico#73] [Ideas]
Hi everyone,

I'm looking to add new features to the project and would love your input! Do you have any ideas or improvements in mind? Share them here! Ludacris requests are also welcomed 😁 

I'll kickstart the list with a few ideas:

- [X] Add animations everywhere to make the app feel more "alive" :) 
- [ ] Add the options to manage recuring transactions and bills. Then on the dashboard show a timeline of your upcming transactions. Maybe even send some notifications as a remainder.
- [x] Implement a desktop optimized UI
- [x] Add support for transaction attachments

**Comment by cioraneanu** [comment]
- Find a way to improve the response time of the Firefly `/api/transactions` endpoint. Hopefully this can be solved by debuging and proposing a change inside FF3 instead of having to create a separate endpoint for it inside Pico.

**Reply by cioraneanu** [reply]
Checked ✅ . Recent improvements in Firefly API response times + the option to have "default list filters" makes this pretty good.

**Comment by cioraneanu** [comment]
**Implement a smart general purpose search.** 
- Should be a single input. 
- Make it support fuzy search
- Search the words in all fields: "description", "tags", "category" etc
- Allow entering numerical values for searching amounts. ex "100" or "<100", "100-120"

**Comment by mvpaderin** [comment]
Hi! Thanks for the great app. One suggestion that I would personally love to see implemented is possibility to add attachments (e.g. PDFs or images of bills / receipts)

**Reply by cioraneanu** [reply]
Added in the `dev` release 😇

**Comment by ElHado** [comment]
Hi, 
your app looks great. I have trouble to get it on my phone because I have no skills in app developement. Can you create an apk for easy install and noobs like me? 
Would be great and makes your app mor accessible. 

Thank you!

**Reply by cioraneanu** [reply]
Thanks! 
But this is a web app, not a mobile native one (apk/ipa). 
All you need is some basic docker knowledge, create a container from this [compose](https://github.com/cioraneanu/firefly-pico/blob/main/docker-compose.pico.yml) and open the app in your browser. After that you can save it to your homescreen because it has PWA support.

**Reply by ElHado** [reply]
OMG 🙄 Looke like an app for me.
Then - I'll give it a try. 👍 Thanks

**Comment by RustyClockwork** [comment]
Hey, I've just discovered this web app. The assistant is a game changer.

I was wondering if it could be somehow made more powerful by allowing multiple tags along with a category.

The input could be something like this, using a comma as separator:
```
My category, My first Tag, My second tag 12.3
```

I also thought of having a special prefix to tell what is what, in case of tags and categories with similar names. This is what I thought, even though I'm not so sure about it:
```
c:My Category, t:My Tag 12.3
```

To be honest, I thought about all of this right before finding out about the templates. I now have a bunch of them to handle the most common type of transactions.

**Reply by cioraneanu** [reply]
I've tried playing with this idea but it felt clunky. On mobile switching between letters and symbols for chars ":", "," was annoying and error prone. 
For my use case the assistant always finds a template or a tag(which already matches to a category) and I almost never have to create unique tag combinations for day to day expenses.

This is a nice concept but I think the changes required are not worth vs the actual user who will use it 😇

**Comment by homaydimsum** [comment]
Amazing web app! Is it possible to attach images/files such as bills, statements, or receipts onto each transaction?

**Reply by cioraneanu** [reply]
Thank you 😇 It is on my todo list #177. Will probably be added in the next major release.

**Comment by cioraneanu** [comment]
Would be awesome to have a nice oboarding process for new users. There could be an option to have a default list of tags / categories / templates auto created for you. Maybe even add this functionality for users who are not new.
